### PR TITLE
chore(storybook): leverage shared types in all existing stories

### PR DESCRIPTION
**Pull request by castastrophe**
## Description

This PR updates existing stories to leverage the shared types provided by the Storybook package. This also adds a new types file `variants.js` for a set of shared global variants (i.e., sizing, quiet, emphasized).

Expect updated sizing drop-down to include full-text labels:

Screenshot 2024-08-06 at 9 25 20 PM

This PR also ran `eslint --fix` on the `.storybook` js assets to allow for whitespace and quotation mark corrections to be applied.

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Validation steps

@marissahuysentruyt 
Expect to see no regressions in knob functionality:
- [x] accordion: size
- [x] actionbar: emphasized
- [x] actionbutton: size, quiet, emphasized, active, static color
- [x] actiongroup: size
- [x] avatar: size, disabled
- [x] badge: size
- [x] breadcrumb: dragged
- [x] button: size, hovered, active, pending, static color
- [x] buttongroup: size
- [x] calendar: padded renamed to isPadded (b/c it's a boolean variable)
- [x] card: quiet
- [x] checkbox: size, emphasized, disabled, checked, indeterminate, read-only
- [x] clearbutton: size, disabled, static color (should still show only white)
- [x] closebutton: size, static color, disabled
- [x] coach indicator: quiet
- [x] combobox: size, quiet, keyboard focused, loading, disabled
- [x] date picker: quiet, disabled, required, read-only
- [x] dial: size, dragged, disabled
- [x] divider: size, static color
- [x] drop zone: dragged
- [x] fieldgroup: required, read only
- [x] fieldlabel: size, disabled, required
- [x] helptext: size, disabled
- [x] icon: size
- [x] infield button: size, quiet, disabled
- [x] link: static color, quiet
- [x] logic button: disabled
- [x] menu: size, disabled, selected
- [x] pagination: size
- [x] picker: size, quiet, keyboard-focused, disabled, loading
- [x] picker button: size, quiet, disabled
- [x] progressbar: size, static color
- [x] progress circle: size, indeterminate, static color
- [x] radio: size, emphasized, checked, disabled, read-only
- [x] rating: emphasized, disabled, read-only
- [x] search: size, quiet, disabled
- [x] slider: size, disabled
- [x] status light: size
- [x] stepper: size, quiet, disabled, keyboard focused
- [x] swatch: size, disabled
- [x] switch: size, emphasized, disabled, checked
- [x] table: size, quiet, emphasized
- [x] tabs: size, quiet, emphasized
- [x] tag: size, emphasized, disabled
- [x] textfield: disabled, keyboard-focused, size, quiet, required, read-only, loading
- [x] thumbnail: size, disabled
- [x] treeview: size, quiet
- [x] typography: size

## To-do list

- [x] I have read the [contribution guidelines](/.github/CONTRIBUTING.md).
- [x] I have updated relevant storybook stories and templates.
- [x] ✨ This pull request is ready to merge. ✨

### CODE REVIEW

File: components/closebutton/stories/template.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ export const Template = ({
 			class=${classMap({
 				[rootClass]: true,
 				[`${rootClass}--size${upperCase(size)}`]: typeof size !== "undefined",
-				[`${rootClass}--static${capitalize(lowerCase(staticColor))}`]:
```
Comment (marissahuysentruyt) [inline components/closebutton/stories/template.js:25]: Great catch here! 😆

**Review by marissahuysentruyt — CHANGES_REQUESTED** [review]
This is looking really great! I had mainly questions, but I found a few things I just wanted to bring up to be sure on!

File: components/actionbutton/stories/actionbutton.stories.js [hunk 2]
```diff
@@ -34,38 +25,13 @@ export default {
...
-			type: { name: "boolean" },
-			table: {
-				type: { summary: "boolean" },
-				category: "Component",
-			},
-			control: "boolean",
-			if: { arg: "isSelected", truthy: true }
-		},
+		isQuiet,
+		isEmphasized,
```
[the remark this answers is not in the document]
Comment (castastrophe) [reply components/actionbutton/stories/actionbutton.stories.js:29]: I'm torn, `isEmphasized` is still true even if not in a selected state.  It's just that the only change to styling happens for it in the selected state right?

File: components/datepicker/stories/template.js
[the remark this answers is not in the document]
Comment (castastrophe) [reply components/datepicker/stories/template.js:1]: Yes good catch! I just updated the PR with this fix.

File: components/fieldgroup/stories/fieldgroup.stories.js [hunk 3]
```diff
@@ -53,8 +53,8 @@ export default {
 		items: { table: { disable: true } },
 		fieldLabel: { table: { disable: true } },
 		helpText: { table: { disable: true } },
-		isRequired: { table: { disable: true } },
-		isReadOnly: { table: { disable: true } },
+		isRequired,
+		isReadOnly,
```
[the remark this answers is not in the document]
Comment (castastrophe) [reply components/fieldgroup/stories/fieldgroup.stories.js:58]: Good call!  Updated.

File: components/progressbar/stories/progressbar.stories.js [hunk 4]
```diff
@@ -9,25 +10,8 @@ export default {
...
-		isIndeterminate: {
-			name: "Indeterminate state",
-			type: { name: "boolean" },
-			table: {
-				type: { summary: "boolean" },
-				category: "State",
-			},
-			control: "boolean",
-		},
+		isIndeterminate: { table: { disable: true } },
```
[the remark this answers is not in the document]
Comment (castastrophe) [reply components/progressbar/stories/progressbar.stories.js:13]: Oh that's my fault - probably an outdated branch issue.

File: components/thumbnail/stories/thumbnail.stories.js [hunk 5]
```diff
@@ -13,29 +13,7 @@ export default {
...
-				"500",
-				"600",
-				"700",
-				"800",
-				"900",
-				"1000",
-			],
-			control: "select",
-		},
+		size: size([50, 75, 100, 200, 300, 400, 500, 600, 700, 800, 900, 1000], false),
```
[the remark this answers is not in the document]
Comment (castastrophe) [reply components/thumbnail/stories/thumbnail.stories.js:16]: It's not mandatory per say, a string or numbers would work but I updated thumbnail to numeric just to be consistent.

File: components/typography/stories/template.js [hunk 6]
```diff
@@ -11,7 +11,7 @@ import "../index.css";
 export const Template = (args = {}, context = {}) => {
 	let {
 		rootClass = "spectrum-Typography",
-		semantics = "body",
+		semantics,
```
[the remark this answers is not in the document]
Comment (marissahuysentruyt) [reply components/typography/stories/template.js:14]: ooooooo ok I see. Got it- thanks.

File: components/actionbutton/stories/actionbutton.stories.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (marissahuysentruyt) [reply components/actionbutton/stories/actionbutton.stories.js:29]: Oh wait- ok so `isEmphasized` is true, even if not in a selected state. Could we add a `description` to the control to call that out somehow?

Screenshot 2024-08-29 at 3 07 41 PM

**Review by marissahuysentruyt — APPROVED** [review]
I'll defer to you on the action button idea! It looks great 🤩 Thanks for making those updates!

File: components/actionbutton/stories/actionbutton.stories.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (marissahuysentruyt) [reply components/actionbutton/stories/actionbutton.stories.js:29]: Maybe you're right...checkbox and radio behave this way, where `isEmphasized` isn't hidden, but you dont' see styling changing until they're checked/selected.

**Pull request merged**